### DISCUSSION TITLE
TSML-UI top scrolling issues

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#822] [Q&A] [answered]
Related to #780 is what I’ve not only observed on my sites, but others in similar ways. 

I ended up taking the suggestion to adjust via CSS, the scroll so it doesn’t cut off the meeting title, but is still cuts off the top widget title that I’d like displayed. 

My preference would have been the way it worked for the legacy plugin. Have Meeting Top widget only, but not have that also apply to the meeting detail. 

You can see how it looks using legacy on our live site. https://sussexaa.org/meetings/

Fiddling with the CSS for scroll has no effect. It either will start from the title Meetings on the list page, or the name of the meeting on the detail.

**Comment by joshreisner** [comment, marked as answer]
im struggling to understand the hostile tone.

theres a lot to say about the differences between fixed headers and static “widget” content and how css can alleviate the first on a case by case basis but the second is a design choice, but it already sounds like you don’t want to hear it. 

i’m also struggling to understand why i would choose to log into this site in the mornings before work - i think i should take a little time off from this community. ❤️

**Reply by Grateful-Viv** [reply]
@joshreisner I am sorry if any hostility was taken. None of that was intended by any means.

I was asking for advice on how to address this and would welcome any suggestions.

I truly appreciate all the help you've given on everything related to this plugin, and this particular question is not big deal. So I'd be grateful we can consider the question dropped.

Edit: After re-reading what I wrote, I can see how it may have been viewed with hostility. Deepest apologies. I am editing the initial question and will make note to self to not go here with any questions early in the morning.

**Reply by joshreisner** [reply]
thanks @Grateful-Viv i appreciate it! likewise i'll make a note to self to log in later in the day once my skin has thickened up a little.

i'll add an item to the TSML UI readme on how to handle the fixed-header overhang issue and link it here. 

once that's done we can discuss what, if anything, can be done to put the meeting-top widget content in view when a user navigates to a meeting

**Reply by Grateful-Viv** [reply]
I think I’ve resolved the scrolling past the meeting name using the CSS you gave on the other thread. 45px was a perfect number for us. And it consistently is good. 

No issues with header or menu bar either. 

And I have a feeling this doesn’t actually bother anyone or even if they notice. In the long run, it’s not that important. The plugin works and people can find their meetings. 

So whenever the widget issue can be dealt with is fine. No rush.

**Reply by joshreisner** [reply]
Thanks Viv. I also just pushed a change a few minutes ago that may fix this problem for other sites. I see better scroll behavior now on https://area59aa.org/meetings/ for example.

**Reply by joshreisner** [reply]
(SEPIA uses their own version so it may take longer for the change to make its way to them)

**Reply by Grateful-Viv** [reply]
When changes are made to TSML UI, is it possible to be alerted to that? Not just this particular issue, but others as well.  I have disabled auto-update for all plugins, and when TSML legacy has an update pending, I always check it out on the staging sites before updating the live site. Usually there's nothing to be concerned about with any updates.

But now I think the CSS on TSML UI

```
* {
  scroll-margin-top: 45px;
}
```

needs to be adjusted or even removed. What is showing is a peak of the widget box when I select a meeting. You can see it here. https://sdiaa-staging-two.sussexaa.org/meetings/

I am so close to using the UI on the live site. 

Thank you!

**Reply by joshreisner** [reply]
sorry about that! so my fix this morning was to calculate the height of any fixed elements and factor that in when scrolling. this didn't take into account your situation, which is an element that only becomes fixed when the user scrolls. this overwrote the css, which looks like you might have removed now? if you don't mind i would put it back but change it to:

```
#tsml-ui {
  scroll-margin-top: 43px;
}
```

regarding your other point, i think what we want to do is have each version of TSML ship with a version of TSML UI. this will lock features to TSML versions, so nothing will get updated on your site unless you update TSML. i'll create a ticket for that.

**Reply by Grateful-Viv** [reply]
It's a tad better using the new CSS, but it's not consistent. Going continually from  <- Back to Meetings to a Meeting still shows some of the widget box. I've tried a few different sizes and it doesn't make a difference. 

I also removed it entirely, and it displays the same.

![Cool-Spring-Serenity](https://user-images.githubusercontent.com/62247421/172479054-76954365-bc51-4453-9543-ac7025f191c8.png)

**Reply by joshreisner** [reply]
i think the case where it will help is if you click from the `/meetings` page before you scroll down, before the header has become fixed

it looks ok to me, i wonder if you are logged into the wp backend when you're seeing the behavior above? if so, is it different from you when you're logged out?

**Reply by Grateful-Viv** [reply]
I checked it on my iPad, iPhone, laptop without being logged in.

Then I logged off the desktop, and it's behaving as it was before you pushed the change and when I had the CSS in. I am sorry I didn't think to log off before before saying anything here.

I also removed `#tsml-ui {
  scroll-margin-top: 43px;
}`
from my global CSS as well as the previous one since it no longer seems needed.

>>i think what we want to do is have each version of TSML ship with a version of TSML UI. this will lock features to TSML versions, so nothing will get updated on your site unless you update TSML. i'll create a ticket for that.

That would awesome.

And again, @joshreisner thank you for everything. I sure do appreciate you.
